update constant module to use scipy value and remove using legacy constants

zhichen3 opened the pull request:
added a script to auto generate scipy constant value. Address issue #1655.

zingale commented:
I think that if we update G here, then the initial models that do self-gravity would need to be regenerated, since Castro would pick up this G.  This is fine, but we should coordinate this carefully.

zhichen3 commented:
> I think that if we update G here, then the initial models that do self-gravity would need to be regenerated, since Castro would pick up this G. This is fine, but we should coordinate this carefully.

Right, the old G is not updated. Only the new G is updated, and that's in comment. So its not being used actually

zhichen3 commented:
I also updated the `burn_cell_metal_chem` unit test script to get the "correct" line number. Previously the test actually results in some integration failure, to see that grep for `fail` in `test_*.out`. With this pr, we no longer see those failure. So the reference line number changed, resulting in comparison between different quantities. Here I updated it and the test passes.

zingale commented:
tests:
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2025-06-07-001/index.html

zingale commented:
I think we will still need to update any Castro initial models that use gamma_law, since the constants are changing.

zingale commented:
oh indeed.  I thought it would be more.  I'll review this today.
